core/src/internal_flags.rs

```diff
@@ -146,11 +175,17 @@ impl InternalFlags {
                     .collect();
                 core.extend(core_since_last_complete.iter());
                 lang.extend(lang_since_last_complete.iter());
+                let (sdk_name, sdk_version) =
+                    if last_sdk_name != sdk_name || last_sdk_version != sdk_version {
+                        (sdk_name.clone(), sdk_version.clone())
+                    } else {
+                        ("".to_owned(), "".to_owned())
+                    };
```

Comment:
I think other SDKs (and mentioned at https://github.com/temporalio/features/issues/321) compare name and version independently, meaning if the version upgraded but not the SDK, only the version is written.

Reply:
Well, that wouldn't be totally independent, since it wouldn't make sense the other way around, but, yeah can do

Reply:
Thanks! I think _technically_ the other SDK impls are independent meaning in some strange world you are able to switch from Go to Java but keep the version number, the version is unchanged.